src/lib.rs

```diff
@@ -113,14 +114,16 @@ fn longest_path(graph: &digraph::PyDiGraph) -> PyResult<Vec<usize>> {
 ///     object must be a DAG without a cycle.
 ///
 /// :returns: The node indices of the longest path on the DAG
-/// :rtype: list
+/// :rtype: NodeIndices
 ///
 /// :raises Exception: If an unexpected error occurs or a path can't be found
 /// :raises DAGHasCycle: If the input PyDiGraph has a cycle
 #[pyfunction]
 #[text_signature = "(graph, /)"]
-fn dag_longest_path(graph: &digraph::PyDiGraph) -> PyResult<Vec<usize>> {
-    longest_path(graph)
```

Comment:
It's easier to leave it returning `Vec<usize>` because `dag_longest_path_length` calls it too. I actually did that in the first rev before pushing this and it errored because there is no `len()` on `NodeIndices` it felt easier to leave it like this and have `dag_longest_path` just move the result into the `NodeIndices` struct as the return and leave it as a `Vec<usize>` for the other method